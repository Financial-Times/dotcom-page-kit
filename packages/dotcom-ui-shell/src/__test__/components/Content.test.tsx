import React from 'react'
import renderer from 'react-test-renderer'
import Subject from '../../components/Content'

describe('dotcom-ui-shell/src/components/Content', () => {
  describe('with stringified contents', () => {
    it('renders the given HTML string without escaping', () => {
      const contents = '<p>Hello, World!</p>'
      const tree = renderer.create(<Subject contents={contents} />).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  describe('with component contents', () => {
    it('renders the components', () => {
      const contents = <p>Hello, World!</p>
      const tree = renderer.create(<Subject contents={contents} />).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })
})
