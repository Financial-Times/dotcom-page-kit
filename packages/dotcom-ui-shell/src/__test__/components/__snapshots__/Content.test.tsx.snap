// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dotcom-ui-shell/src/components/Content with component contents renders the components 1`] = `
<p>
  Hello, World!
</p>
`;

exports[`dotcom-ui-shell/src/components/Content with stringified contents renders the given HTML string without escaping 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "<p>Hello, World!</p>",
    }
  }
  style={
    Object {
      "display": "contents",
    }
  }
/>
`;
